tests/cupy_tests/test_cudnn.py

```diff
@@ -343,3 +343,115 @@ def test_call(self):
+    'ksize': [1, 3, 5],
+    'stride': [1, 2, 4],
+    'auto_tune': [True, False],
+    'layout': layouts
+}))
+@unittest.skipUnless(cudnn_enabled, 'cuDNN is not available')
+class TestConvolutionInvalid(unittest.TestCase):
+
+    def setUp(self):
+        self._workspace_size = cudnn.get_max_workspace_size()
+        if libcudnn.getVersion() < 7500:
+            self.skip = True
+            self.skip_msg = "cuDNN version is older than 7.5.0"
+            return
+        else:
+            self.skip = False
+        n = 16
+        x_c, y_c = 64, 64
+        x_h, x_w = 32, 32
+        y_h, y_w = x_h // self.stride, x_w // self.stride
+        self.pad = (self.ksize - 1) // 2
+        if self.layout == libcudnn.CUDNN_TENSOR_NHWC:
+            x_shape = (n, x_h, x_w, x_c)
+            y_shape = (n, y_h, y_w, y_c)
+            W_shape = (y_c, self.ksize, self.ksize, x_c)
+        else:
+            x_shape = (n, x_c, x_h, x_w)
+            y_shape = (n, y_c, y_h, y_w)
+            W_shape = (y_c, x_c, self.ksize, self.ksize)
+        self.x = cupy.ones(x_shape, dtype=self.dtype)
+        self.W = cupy.ones(W_shape, dtype=self.dtype)
+        self.y = cupy.empty(y_shape, dtype=self.dtype)
+        self.gx = cupy.empty(x_shape, dtype=self.dtype)
+        self.gW = cupy.empty(W_shape, dtype=self.dtype)
+        self.gy = cupy.ones(y_shape, dtype=self.dtype)
+        cudnn.set_max_workspace_size(0)
+
+    def tearDown(self):
+        cudnn.set_max_workspace_size(self._workspace_size)
+
+    def call_forward(self):
+        cudnn.convolution_forward(
+            self.x, self.W, None, self.y,
+            pad=(self.pad, self.pad), stride=(self.stride, self.stride),
+            dilation=(1, 1), groups=1,
+            auto_tune=self.auto_tune, tensor_core="always",
+            d_layout=self.layout, w_layout=self.layout)
+
+    def test_forward(self):
+        if self.skip:
+            return unittest.SkipTest(self.skip_msg)
+        self.call_forward()
+
+    def call_backward_filter(self):
+        cudnn.convolution_backward_filter(
+            self.x, self.gy, self.gW,
+            pad=(self.pad, self.pad), stride=(self.stride, self.stride),
+            dilation=(1, 1), groups=1, deterministic=0,
+            auto_tune=self.auto_tune, tensor_core="always",
+            d_layout=self.layout, w_layout=self.layout)
+
+    def test_backward_filter(self):
+        if self.skip:
+            return unittest.SkipTest(self.skip_msg)
+        err = None
+        if (self.layout == libcudnn.CUDNN_TENSOR_NHWC and
+                self.dtype == numpy.float64):
+            if self.auto_tune:
+                err = RuntimeError
+            else:
+                err = libcudnn.CuDNNError
+        # Notice that the error above may not occur with cuDNN newer than 7.6
+        if err is not None:
+            with self.assertRaises(err):
+                self.call_backward_filter()
+        else:
+            self.call_backward_filter()
```

Comment:
Ditto, is this testing invalid cases? I think it can be skipped otherwise, assuming that the positive behavior is already covered in other tests but I may be wrong.

Reply:
I'm also wondering which is better, to test only invalid cases or to test both invalid and valid cases. But I'm rather inclined it is better to test both cases, since it looks that existing tests only cover cases when kernel size is 3 and stride is 2.

Reply:
Ah, I see. Can't we parameterize the existing valid tests instead?

Reply:
I think that is good to increase test cases of existing test, but we should do it at another PR.

Reply:
That sounds good to me. Could you possibly write a TODO comment in the existing tests to do this?

Reply:
Got it. comment added.